docs/proposals/Automated-Certificate-and-Key-Updates.md

```diff
@@ -0,0 +1,212 @@
+- HCP Vault: https://developer.hashicorp.com/vault/docs/what-is-vault
+- GCP KMS: https://cloud.google.com/kms/docs/key-management-service
+- AWS Signer: https://docs.aws.amazon.com/signer/latest/developerguide/Welcome.html
+- Azure Trusted Signing: https://learn.microsoft.com/en-us/azure/trusted-signing/overview
+
+Here’s a comparison of various KMSs used for creating and managing keys/certificates for signing and verification:
+
+|                            | **AWS KMS** | **Azure Key Vault** | **HCP Vault** | **GCP KMS** |
+|----------------------------|-------------|---------------------|---------------|-------------|
+| **Key Operations**         | Enable, Disable, Delete, Versioning, Rotate | Enable, Disable, Delete, Versioning, Rotate | Enable, Disable, Delete, Versioning, Rotate | Enable, Disable, Delete, Versioning, Rotate |
+| **Key Rotation**           | Manual for asymmetric keys | Automatic/Manual | Automatic/Manual | Manual for asymmetric keys |
+| **Rotation Result**        | New version created, old version remains valid | New version created, old version remains valid | New version created, old version remains valid | New version created, old version remains valid |
+| **Events**                 | CloudWatch Events for key state changes | Azure Event Grid for key state changes | Custom implementation (e.g., using Vault's audit logs) | Cloud Pub/Sub for key state changes |
+| **Certificate management** | Not supported | Supported | Supported | Not supported |
+
+Additionally, here’s a comparison of fully managed code-signing services related to Ratify signature verification scenarios. The difference from KMS in previous table is that fully managed code-signing services do not require users to create keys/certificates, instead they managed the key/certificates for users and normally these key/certificates have short validity. For fully managed code-signing, normally users set up a profile for signing purpose. The profile identifier is similar to the key identifier that used by KMS in previous table. Users can revoke the profile, so that it cannot be used for signing an verification. 
+
+|                        | **AWS Signer**                                   | **Azure Trusted Signing**             |
+|------------------------|--------------------------------------------------|---------------------------------------|
+| **Configuration**      | Signing profile                                  | Certificate profile                   | 
+| **Revocation**         | Signature revocation, signing profile revocation | Certificate profile revocation        |
+| **Signature Type**     | Notary Project signature                         | Notary Project signature              |
+
+Lastly, here’s a comparison of `KMP` resources used for various KMPs in Ratify v1.2.0:
+
+|                                       | **AWS KMS** | **Azure KV** | **HCP Vault** | **GCP KMS** | **AWS Signer** | **Azure Trusted Signing** |
```

Comment:
clarification, can Customer use HCP GCP with ratify today?

Reply:
@susanshi  Just update the text to make it more clear.